### PR TITLE
Add automatic meta.date management to JwstDataModel

### PR DESCRIPTION
**Pull request by eslavich**
To support Roman storing astropy.time.Time objects in meta.date, the code that automatically maintains that attribute will be removed from stdatamodels starting in version 0.2.0 (not yet released).  This PR re-implements that feature in JwstDataModel using a new hook that we added to the base DataModel class.

This PR should *not* be merged before jwst 0.18.0 is released, since stdatamodels 0.2.0 will include other changes that may cause trouble for jwst.  I'm mainly opening the PR now so I don't forget about it.

### CODE REVIEW

**Review by jdavies-st — COMMENTED** [review]
Will this change allow us to unpin `stdatamodels~=0.1.0`?  If so, we should do it here in this PR.  Though maybe there will be other stuff in 0.2.0 that is unaddressed here?

**Comment by jdavies-st** [comment]
Just a note here that the changes over at `stdatamodels` to allow for an astropy time object in `meta.date` is causing our `jwst` nightly regression tests with dev dependencies (including dev `stdatamodels`) to fail, but that is expected.

https://plwishmaster.stsci.edu:8081/job/RT/job/JWST_UNSTABLE/427/

Those will fail until this PR is merged.

**Review by jdavies-st — COMMENTED** [review]
It's probably OK to now get this PR to a state to be merged, now that 0.18.x is done.

The minimum will be unpinning `stdatamodels` in setup.cfg

https://github.com/spacetelescope/jwst/blob/3914b95672188f700b2b21a6057e5e52627a56c4/setup.cfg#L45

Might be good to run some regression tests as well after rebasing.

**Comment by eslavich** [comment]
Regression test run on this branch with the stdatamodels release candidate: https://plwishmaster.stsci.edu:8081/job/RT/job/eslavich-dev/63/

File: JenkinsfileRT
Comment (jdavies-st) [inline JenkinsfileRT:76]: Removing this means the default conda version will be used, which might be out-of-date.  We should probably just bump this to the latest for now until we fix the Jenkins CI helpers to get the latest, i.e.

https://github.com/spacetelescope/jenkins_shared_ci_utils/blob/37606ed9c9a9133e7983183e52976683d8d25ece/vars/utils.groovy#L106-L108

**Review by jdavies-st — APPROVED** [review]

File: JenkinsfileRT
Comment (eslavich) [reply JenkinsfileRT:76]: Oops, yes, I have a PR open over there (https://github.com/spacetelescope/jenkins_shared_ci_utils/pull/80) but I agree we should wait until that's actually merged.

**Pull request merged**
